build.gradle.kts

```diff
@@ -1,70 +1,109 @@
-import kotlinx.validation.ApiValidationExtension
-import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
-import org.jlleitschuh.gradle.ktlint.KtlintExtension
-import org.jlleitschuh.gradle.ktlint.reporter.ReporterType
+plugins {
+  kotlin("multiplatform") version "1.6.20"
+  id("com.android.library")
+}
 
-println("Building with Kotlin compiler version ${Versions.KotlinCompiler}")
+group = "logcatx"
+version = "1.0-SNAPSHOT"
 
-buildscript {
-  repositories {
+repositories {
     mavenCentral()
     gradlePluginPortal()
     google()
-    // For binary compatibility validator.
-    maven { url = uri("https://kotlin.bintray.com/kotlinx") }
-  }
-
-  dependencies {
-    classpath(Dependencies.Build.Android)
-    classpath(Dependencies.Build.MavenPublish)
-    classpath(Dependencies.Build.Kotlin)
-    classpath(Dependencies.Build.Ktlint)
-    classpath(Dependencies.Build.BinaryCompatibility)
-  }
 }
 
-// We use JetBrain's Kotlin Binary Compatibility Validator to track changes to our public binary
-// APIs.
-// When making a change that results in a public ABI change, the apiCheck task will fail. When this
-// happens, run ./gradlew apiDump to generate updated *.api files, and add those to your commit.
-// See https://github.com/Kotlin/binary-compatibility-validator
-apply(plugin = "binary-compatibility-validator")
+kotlin {
+    jvm {
+        compilations.all {
+            kotlinOptions.jvmTarget = "1.8"
+        }
+//        withJava()
+//        testRuns["test"].executionTask.configure {
+//            useJUnitPlatform()
+//        }
+    }
+    js(IR) {
+        browser {
+            commonWebpackConfig {
+                cssSupport.enabled = true
+            }
+        }
+        binaries.executable()
+    }
+    macosX64("native")  {
+        binaries {
+            sharedLib {
+                baseName = "logcat" // on Linux and macOS
+                // baseName = "libnative" // on Windows
```

Comment:
what's this about?

Reply:
This has been cleaned up.  I also added iOS framework generation to see if that works.  Seems to create a Framework successfully, though I'm not an iOS dev so I'm not sure how well it actually works.  I might try to recreate your sample project in Swift if I can find some time.